File: Codewars/Bowling Pins.cs
// Bowling Pins
// https://www.codewars.com/kata/bowling-pins/
// by Equim on 17-02-24

namespace BowlingPins {
    using System;
    using System.Text;
    using System.Collections.Generic;
    using System.Linq;

    public class Bowling
    {
        public string BowlingPins(int[] arr)
        {
            if (arr.Length == 0)
               return "I I I I\n I I I \n  I I  \n   I   ";
            Array.Sort(arr);
            StringBuilder[] builder = new String[] { "", " ", "  ", "   " }.Select(p => new StringBuilder(p)).ToArray();
            for (int i = 1; i <= 10; i++)
            {
                int line = i >= 7 ? 0 : i >= 4 ? 1 : i >= 2 ? 2 : 3;
                builder[line].Append(Array.Exists(arr, p => p == i) ? "  " : "I ");
            }
            return String.Join("\n", builder.Select(p => p.ToString().TrimEnd().PadRight(7, ' ')));
        }
    }
}

File: Codewars/Bowling Pins.py
# Bowling Pins
# https://www.codewars.com/kata/bowling-pins/
# by Equim on 17-02-24

def bowlingPins(arr):
    if len(arr) == 0:
        return "I I I I\n I I I \n  I I  \n   I   "
    arr.sort()
    builder = ["", " ", "  ", "   "]
    for i in range(1, 11):
        line = 0 if i >= 7 else 1 if i >= 4 else 2 if i >= 2 else 3
        builder[line] += "  " if i in arr else "I "
    return '\n'.join([x.rstrip(' ').ljust(7) for x in builder])
